packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -697,7 +698,39 @@ export function makeVirtualObjectManager(
     const durableKindDescriptor = kindDescriptors.get(kindHandle);
     assert(durableKindDescriptor);
     const { kindID, tag } = durableKindDescriptor;
-    return defineKindInternal(kindID, tag, init, actualize, finish, true);
+    const maker = defineKindInternal(
+      kindID,
+      tag,
+      init,
+      actualize,
+      finish,
+      true,
+    );
+    definedDurableKinds.add(kindID);
+    return maker;
+  }
+
+  function insistAllDurableKindsReconnected() {
+    // identify all user-defined durable kinds by iterating `vom.kind.*`
+    const start = `vom.kind.`;
+    const end = `vom.kind/`; // '.' and '/' are sequential in ASCII/UTF-8
+    let key = '';
+    let value;
+    const missing = [];
+    for (;;) {
+      [key, value] = syscall.vatstoreGetAfter(key, start, end);
```

Comment:
Don't manage `start` and `end` yourself.  The API is designed to handle prefix encoding so you can just do
```
      [key, value] = syscall.vatstoreGetAfter(key, 'vom.kind.');
```
Let the vatstore worry about how ASCII works, 'cause it's engineered to do that.

A better idiom for the loop as a whole is actually:
```
let [key, value] = syscall.vatstoreGetAfter('', prefix);
while (key) {
  <<do stuff>>
  [key, value] = syscall.vatstoreGetAfter(key, prefix);
}
```